### PR TITLE
Add support for checked stake and vote instructions

### PR DESCRIPTION
**Pull request by CriesofCarrots**

### CODE REVIEW

File: libsol/stake_instruction.c [hunk 1]
```diff
@@ -95,6 +99,27 @@ static int parse_stake_initialize_instruction(
...
+    instruction_accounts_iterator_init(&it, header, instruction);
+
+    BAIL_IF(instruction_accounts_iterator_next(&it, &info->account));
+    // Skip rent sysvar
+    BAIL_IF(instruction_accounts_iterator_next(&it, NULL));
+    BAIL_IF(instruction_accounts_iterator_next(&it, &info->stake_authority));
+    BAIL_IF(instruction_accounts_iterator_next(&it, &info->withdraw_authority));
+
+    // No lockup on checked instructions
+    info->lockup. present = StakeLockupHasNone;
```
Comment (t-nelson) [inline libsol/stake_instruction.c]: I was going to suggest that these stray spaces be removed throughout, but I see now that I introduced them :sweat_smile: 

Must've got carried away with a find/replace at some point :thinking:

**Review by t-nelson — APPROVED** [review]
Looks great!  I'm glad someone else can still move around in here :slightly_smiling_face:

File: libsol/stake_instruction.c (hunk 1 above)
Comment (CriesofCarrots) [reply libsol/stake_instruction.c]: Haha, I thought it was a stylistic choice ;) There are just a handful of them, so I'll clean up

**Comment by CriesofCarrots** [comment]
Will wait to bump this to solana v1.7.6
Also, looks like the spl bumps are not needed 🎉

**Comment by t-nelson** [comment]
Merged to upstream via https://github.com/LedgerHQ/app-solana/pull/9

**Pull request closed without merging**